Flaky test: HttpParallelRequest_3_IT fails with ConcurrentModificationException

## Description

`HttpParallelRequest_3_IT` (executed as `HttpServerIT.parallelRequestsIT`) is flaky in GitHub Actions. It fails intermittently with a `ConcurrentModificationException` caused by concurrent access to the `RestTemplate` interceptors list during parallel HTTP request sending.

## Steps to Reproduce

1. Run `HttpParallelRequest_3_IT` in CI (GitHub Actions) — the test sends multiple HTTP requests in parallel
2. The failure is intermittent and depends on thread scheduling / timing

## Expected Behavior

Parallel HTTP requests should be sent without thread-safety issues.

## Actual Behavior

The test fails with:

```
org.citrusframework.exceptions.CitrusRuntimeException: java.util.ConcurrentModificationException
    at org.citrusframework.actions.SendMessageAction.lambda$doExecute$1(SendMessageAction.java:165)
    ...
Caused by: java.util.ConcurrentModificationException
    at java.base/java.util.ArrayList.sort(ArrayList.java:1723)
    at org.springframework.core.annotation.AnnotationAwareOrderComparator.sort(AnnotationAwareOrderComparator.java:109)
    at org.springframework.http.client.support.InterceptingHttpAccessor.setInterceptors(InterceptingHttpAccessor.java:70)
    at org.citrusframework.http.client.HttpEndpointConfiguration.getRestTemplate(HttpEndpointConfiguration.java:240)
    at org.citrusframework.http.client.HttpClient.send(HttpClient.java:133)
    at org.citrusframework.actions.SendMessageAction.lambda$doExecute$1(SendMessageAction.java:160)
```

## Additional Context

Root cause: When multiple threads send HTTP requests in parallel using the same `HttpClient`, each call to `HttpClient.send()` invokes `HttpEndpointConfiguration.getRestTemplate()`, which calls `setInterceptors()` on the shared `RestTemplate`. Spring's `InterceptingHttpAccessor.setInterceptors()` sorts the interceptors list in place via `AnnotationAwareOrderComparator.sort()`. When two threads hit this concurrently, the underlying `ArrayList.sort()` throws `ConcurrentModificationException`.

This is not a test-only issue — any production use of parallel sends through a shared `HttpClient` could trigger the same race condition.

--- a/endpoints/citrus-http/src/main/java/org/citrusframework/http/client/HttpEndpointConfiguration.java
+++ b/endpoints/citrus-http/src/main/java/org/citrusframework/http/client/HttpEndpointConfiguration.java
@@ -154,7 +154,7 @@ public class HttpEndpointConfiguration extends AbstractPollableEndpointConfigura
      */
     public void setRestTemplate(RestTemplate restTemplate) {
         clientInterceptors.addAll(restTemplate.getInterceptors());
-        restTemplate.setInterceptors(clientInterceptors);
+        restTemplate.setInterceptors(new ArrayList<>(clientInterceptors));
         this.restTemplate = restTemplate;
     }
 
@@ -237,7 +237,7 @@ public class HttpEndpointConfiguration extends AbstractPollableEndpointConfigura
     public RestTemplate getRestTemplate() {
         if (restTemplate == null) {
             restTemplate = new RestTemplate();
-            restTemplate.setInterceptors(clientInterceptors);
+            restTemplate.setInterceptors(new ArrayList<>(clientInterceptors));
         }
 
         restTemplate.setRequestFactory(getRequestFactory());
@@ -283,7 +283,12 @@ public class HttpEndpointConfiguration extends AbstractPollableEndpointConfigura
      */
     public void setClientInterceptors(List<ClientHttpRequestInterceptor> clientInterceptors) {
         this.clientInterceptors = clientInterceptors;
-        getRestTemplate().setInterceptors(clientInterceptors);
+
+        if (restTemplate == null) {
+            getRestTemplate();
+        } else {
+            restTemplate.setInterceptors(new ArrayList<>(clientInterceptors));
+        }
     }
 
     /**

--- a/endpoints/citrus-http/src/test/java/org/citrusframework/http/client/HttpClientTest.java
+++ b/endpoints/citrus-http/src/test/java/org/citrusframework/http/client/HttpClientTest.java
@@ -18,10 +18,17 @@ package org.citrusframework.http.client;
 
 import java.net.URI;
 import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Random;
+import java.util.concurrent.CyclicBarrier;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
 
 import org.citrusframework.endpoint.resolver.EndpointUriResolver;
 import org.citrusframework.http.message.HttpMessage;
+import org.citrusframework.http.interceptor.LoggingClientInterceptor;
 import org.citrusframework.http.message.HttpMessageHeaders;
 import org.citrusframework.message.DefaultMessage;
 import org.citrusframework.message.ErrorHandlingStrategy;
@@ -516,6 +523,35 @@ public class HttpClientTest extends AbstractTestNGUnitTest {
         Assert.assertEquals(responseMessage.getReasonPhrase(), "OK");
 
         verify(restTemplate).setInterceptors(anyList());
+    }
+
+    @Test
+    public void testConcurrentGetRestTemplate() throws Exception {
+        HttpEndpointConfiguration endpointConfiguration = new HttpEndpointConfiguration();
+
+        int threadCount = 10;
+        CyclicBarrier barrier = new CyclicBarrier(threadCount);
+        ExecutorService executor = Executors.newFixedThreadPool(threadCount);
+        try {
+            List<Future<RestTemplate>> futures = new ArrayList<>();
+            for (int i = 0; i < threadCount; i++) {
+                futures.add(executor.submit(() -> {
+                    barrier.await();
+                    return endpointConfiguration.getRestTemplate();
+                }));
+            }
+
+            for (Future<RestTemplate> future : futures) {
+                RestTemplate template = future.get();
+                Assert.assertNotNull(template);
+                Assert.assertFalse(template.getInterceptors().isEmpty());
+                Assert.assertTrue(template.getInterceptors().stream()
+                        .anyMatch(LoggingClientInterceptor.class::isInstance));
+            }
+        } finally {
+            executor.shutdown();
+        }
+
     }
 
     @Test

